.github/workflows/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+name: 'Shinyapps Package Dependency Tests'
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'packages/**'
+      - '.github/workflows/*.yml'
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  list-tests:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.list-tests.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Print Tests
+        run: echo "matrix=$(ls -1 -d packages/* | grep -v gsl | tr '\n' '\0' | xargs -0 -n 1 basename | jq -Rsc '. / "\n" - [""]')"
+      - name: List Tests As Output
+        id: list-tests
+        run: echo "matrix=$(ls -1 -d packages/* | grep -v gsl | tr '\n' '\0' | xargs -0 -n 1 basename | jq -Rsc '. / "\n" - [""]')" >> $GITHUB_OUTPUT
```

Comment:
alternatively you could pipe this to `tee -a $GITHUB_OUTPUT` to roll up these two steps